File: CommonCoreLegacy/src/main/java/com/skedgo/tripkit/common/agenda/IRealTimeElement.java
package com.skedgo.tripkit.common.agenda;

/**
 * Signifies a class is able to fetch timetable information
 */
public interface IRealTimeElement {

    String getStartStopCode();

    void setStartStopCode(String startStopCode);

    String getEndStopCode();

    void setEndStopCode(String endStopCode);

    String getServiceTripId();

    String getOperator();

    long getStartTimeInSecs();
}

File: CommonCoreLegacy/src/main/java/com/skedgo/tripkit/common/agenda/IRealTimeElement.kt
package com.skedgo.tripkit.common.agenda

/**
 * Signifies a class is able to fetch timetable information
 */
interface IRealTimeElement {
    var startStopCode: String?
    var endStopCode: String?
    val serviceTripId: String?
    val operator: String?
    val startTimeInSeconds: Long
}